src/daft-recordbatch/src/ops/window_states/mod.rs

```diff
@@ -0,0 +1,178 @@
+        {
+            Ok(v) => v,
+            Err(_) => return Some(Ordering::Less),
+        };
+
+        let other_value = match other
+            .source
+            .slice(other.idx as usize, (other.idx as usize) + 1)
+        {
+            Ok(v) => v,
+            Err(_) => return Some(Ordering::Greater),
+        };
+
+        if self_value.data_type() == &DataType::Float64
+            && other_value.data_type() == &DataType::Float64
+        {
+            let self_val_opt = self_value
+                .downcast::<Float64Array>()
+                .ok()
+                .and_then(|arr| arr.get(0));
+            let other_val_opt = other_value
+                .downcast::<Float64Array>()
+                .ok()
+                .and_then(|arr| arr.get(0));
+
+            match (self_val_opt, other_val_opt) {
+                (Some(self_val), Some(other_val)) => {
+                    if self_val.is_nan() || other_val.is_nan() {
+                        return if self_val.is_nan() && other_val.is_nan() {
+                            Some(self.idx.cmp(&other.idx))
+                        } else if self_val.is_nan() {
+                            Some(Ordering::Greater)
+                        } else {
+                            Some(Ordering::Less)
+                        };
+                    }
+                }
+                (None, None) => return Some(self.idx.cmp(&other.idx)),
+                (None, Some(_)) => return Some(Ordering::Less),
+                (Some(_), None) => return Some(Ordering::Greater),
+            }
+        }
+
+        match self_value.lt(&other_value) {
+            Ok(result) => {
+                if result.get(0).unwrap_or(false) {
+                    return Some(Ordering::Less);
+                }
+            }
+            Err(_) => return None,
+        }
+
+        match self_value.equal(&other_value) {
+            Ok(result) => {
+                if result.get(0).unwrap_or(false) {
+                    return Some(self.idx.cmp(&other.idx));
+                }
+            }
+            Err(_) => return None,
+        }
+
+        Some(Ordering::Greater)
+    }
+}
+
+impl Ord for IndexedValue<'_> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.partial_cmp(other).unwrap_or(Ordering::Equal)
+    }
+}
+
+pub fn create_window_agg_state<'a>(
+    source: &'a Series,
+    agg_expr: &AggExpr,
+    total_length: usize,
+) -> DaftResult<Option<Box<dyn WindowAggStateOps<'a> + 'a>>> {
+    match agg_expr {
+        AggExpr::Sum(_) => sum::create_for_type(source, total_length),
+        AggExpr::Count(_, mode) => Ok(Some(Box::new(CountWindowState::with_count_mode(
+            source,
+            total_length,
+            *mode,
+        )))),
+        AggExpr::Min(_) => Ok(Some(Box::new(MinWindowState::new(source, total_length)))),
+        AggExpr::Max(_) => Ok(Some(Box::new(MaxWindowState::new(source, total_length)))),
```

Comment:
The NaN treatment of min and max is kind of unclear to me, even for the aggregation kernels right now. Maybe let's just comment these out for now until we solidify the behavior of NaNs in min and max, with a comment describing why.